setup.py

```diff
@@ -23,62 +23,101 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
-from setuptools import setup
+import sys
+import logging
+
+from setuptools import setup, find_packages
 from setuptools.extension import Extension
-from Cython.Build import cythonize
-import Cython.Compiler.Options as CCO
 from setuptools.command.install import install
-import numpy as np          
 
-CCO.buffer_max_dims = 9
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+log = logging.getLogger()
 
-class custom_install(install):
-    def run(self):
-        os.system("python setup.py build_ext --inplace -f")
-        install.run(self)
+# Courtesy of rasterio's setup.py.
+
+try:
+    import numpy as np
+except ImportError:
+    log.critical("Numpy and its headers are required to run setup(). Exiting.")
+    sys.exit(1)
```

Comment:
don't raise exceptions or exit manually in the setup script.  It should still be possible to run the setup.py script without any of the dependencies installed.

Reply:
I am not sure I understand. Surely installation will fail if you have unmet dependencies?

Reply:
well how as somebody going to be able to say what the dependencies are if the setup.py script fails to parse if the dependencies are not installed?